js/boaexchange.js

```diff
@@ -0,0 +1,1021 @@
+            result[code] = {
+                'id': id,
+                'code': code,
+                'address': address,
+                'info': currency,
+                'type': undefined,
+                'name': currency['name'],
+                'active': currency['maintenance'],
+                'fee': fee,
+                'precision': precision,
+                'limits': {
+                    'amount': {
+                        'min': Math.pow (10, -precision),
+                        'max': Math.pow (10, precision),
+                    },
+                    'price': {
+                        'min': Math.pow (10, -precision),
+                        'max': Math.pow (10, precision),
+                    },
+                    'cost': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'withdraw': {
+                        'min': fee,
+                        'max': Math.pow (10, precision),
+                    },
+                },
+            };
+        }
+        return result;
+    }
+
+    parseDepositAddress (depositAddress, currency = undefined) {
+        let address = this.safeString (depositAddress['address'], 'address');
+        const code = this.commonCurrencyCode (this.safeString (depositAddress['coin'], 'code'));
+        this.checkAddress (address);
+        return {
+            'currency': code,
+            'address': address,
+            'tag': undefined,
+            'info': depositAddress,
+        };
+    }
+
+    parseFundingLimits (limits, symbol = undefined, params = {}) {
+        return {
+            'info': limits,
+            'min': this.safeFloat (limits['min']),
+            'max': this.safeFloat (limits['max']),
+        };
+    }
+
+    parseLedgerEntries (entries, currency = undefined) {
+        let results = [];
+        for (let i = 0; i < entries.length; i++) {
+            results.push (this.parseLedgerEntry (entries[i], currency));
+        }
+        return results;
+    }
+
+    parseLedgerEntry (entry, currency = undefined) {
+        let direction = undefined;
+        const id = this.safeString (entry, 'id');
+        let type = this.safeString (entry, 'ledger_type');
+        if (type === 'pre_order' || type === 'order' || type === 'order_cancel' || type === 'order_return') {
+            type = 'trade';
+        } else if (type === 'trade_fee') {
+            type = 'fee';
+        } else if (type === 'trade_referral') {
+            type = 'referral';
+        } else if (type === 'airdrop') {
+            type = 'cashback';
+        } else if (type !== 'trade') {
+            type = 'transaction';
+        }
+        const code = this.safeCurrencyCode (entry['coin'], 'code', currency);
+        let amount = this.safeFloat (entry, 'amount');
+        if (amount < 0) {
+            direction = 'out';
+        } else {
+            direction = 'in';
+        }
```

Comment:
After this, the amount has to be an absolute value, so we have to add `amount = Math.abs (amount);` here.

Reply:
done